src/main/java/org/openRealmOfStars/AI/PathFinding/PathPoint.java

```diff
@@ -90,4 +90,11 @@ public boolean equals(Object obj) {
     }
     return false;
   }
+
+  @Override
+  public int hashCode() {
+    int result = x;
+    result = 31 * result + y;
```

Comment:
This hashcode is not unique. For example X,Y = (1,0) is equal to (0,31)
Wouldn't at the moment maximum map size is 256x256 so if multiplier would be bigger than that I think this would be then unique enough.